### DISCUSSION TITLE
API key

### DISCUSSION THREAD

**Discussion by rdalen** [Part-DB/Part-DB-server#548] [Q&A] [answered]
I generated a API key for KiCad with an expiration date of 01-01-2100 - but now i get an "Internal Server Error (Status 500)" when i try to open the User settings menu again. Did I perhaps enter an illegal date for Part-DB? Is there a backdoor to fix this?
The Docker error log says;
[error log.md](https://github.com/Part-DB/Part-DB-server/files/14531302/error.log.md)

**Comment by jbtronics** [comment, marked as answer]
There seems to be some kind of year 2038 problem with date formatting on 32 bit systems, which cause this behavior. I'm not sure if there is an adequate fix, besides restricting dates input before 2038 (which is not really a solution in the long term).

To solve your problem it might be the easiest way, to access the MySQL or SQLite database table directly and delete the entry for that key from the `api_tables` table. 

This should be possible by executing `php bin/console doctrine:query:sql "DELETE FROM api_keys"` in the docker container (attention this deletes all currently defined API keys).

**Reply by rdalen** [reply]
Thnx for your quick response! 
Your solution gave an error initially because the table is called api-tokes instead of api-keys / api-tables ;-) 
But after a second try (in my case `docker exec --user=www-data partdb php bin/console doctrine:query:sql "DELETE FROM api_tokens"` because I'm running it in a dockercontainer) I'm now up&running again and can go over at least 10 years with a newly generated API key :-)
Many thnx!
